### PR TITLE
fix: add withdrawals

### PR DESCRIPTION
**Pull request by greged93**
Time spent on this PR: 0.1 day

Resolves: #824 

# Pull Request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?
Empty withdrawals are added to the Block. Invariant also added to check the withdrawal_root is always the empty hash.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/eth_provider/constant.rs [hunk 1]
```diff
@@ -1,7 +1,11 @@
 use lazy_static::lazy_static;
+use reth_primitives::B256;
+use std::str::FromStr;
 
 lazy_static! {
     pub static ref MAX_PRIORITY_FEE_PER_GAS: u64 = 0;
+    pub static ref EMPTY_HASH: B256 =
```
Comment (Eikix) [inline src/eth_provider/constant.rs:7]: can we not import it from reth?

**Comment by Eikix** [comment]
do you think we should create a rust feature called invariants and all a few of them in a subsequent issue? or it's overkill

**Comment by greged93** [comment]
I think that's a bit overkill indeed

File: src/eth_provider/constant.rs [hunk 2]
```diff
@@ -18,7 +14,7 @@ use {
     },
     starknet_crypto::FieldElement,
     std::sync::Arc,
-    std::{env::var, sync::OnceLock},
+    std::{env::var, str::FromStr, sync::OnceLock},
```
Comment (Eikix) [inline src/eth_provider/constant.rs]: do we need this FromStr?

Comment (greged93) [reply src/eth_provider/constant.rs]: Yes because of the `FieldElement::from_str`, if you check it was actually there before

**Review by Eikix — APPROVED** [review]
lgtm

**Pull request merged**